app/javascript/contentDisplayPolicy/initHiddenComments.js

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable no-alert */
+export default function initHiddenComments() {
+  function hide(commentId) {
+    const confirmMsg = `
+    Are you sure you want to hide this comment? This will hide the comment in your post, but will still be visible in the comment's permalink.
+
+    You can also consider blocking the person or reporting abuse.
+    `;
+    const confirmHide = window.confirm(confirmMsg)
+    if(confirmHide) {
+      fetch(`/comments/${commentId}/hide`, {
+        method: 'PATCH',
+        headers: {
+          'X-CSRF-Token': window.csrfToken,
+        },
+      })
+        .then(response => response.json())
+        .then(response => {
+          if (response.hidden === 'true') {
+            /* eslint-disable-next-line no-restricted-globals */
+            location.reload();
+          }
+        });
+    }
+  }
+
+  function unhide(commentId) {
+    fetch(`/comments/${commentId}/unhide`, {
+      method: 'PATCH',
+      headers: {
+        'X-CSRF-Token': window.csrfToken,
+      },
+    })
+      .then(response => response.json())
+      .then(response => {
+        if (response.hidden === 'false') {
+          /* eslint-disable-next-line no-restricted-globals */
+          location.reload();
```

Comment:
Using `location.reload()` feels like a hack to me.

Reply:
We could either disappear the comment (if it's top level) and replace the comment with the [hidden] version if not. But I don't think `location.reload()` is so bad either. This won't be used super often and reloading is a clean way to ensure it works properly.